### PR TITLE
[Visualizer] Added unit test 

### PR DESCRIPTION
**Pull request by larshg**
Where points get removed and updatepointcloud is called on visualizer.

Basically @maxfleur MCVE from [here](https://github.com/PointCloudLibrary/pcl/issues/4001#issuecomment-622439905).

### CODE REVIEW

File: test/visualization/test_visualization.cpp [hunk 1]
```diff
@@ -60,6 +60,36 @@ PointCloud<PointNormal>::Ptr cloud_with_normals1 (new PointCloud<PointNormal>);
 search::KdTree<PointXYZ>::Ptr tree3;
 search::KdTree<PointNormal>::Ptr tree4;
 
+pcl::PointCloud<pcl::PointXYZRGB>::Ptr getPointCloud(int pointNumber) {
+  pcl::PointCloud<pcl::PointXYZRGB>::Ptr cloud(new pcl::PointCloud<pcl::PointXYZRGB>());
+
+  for (int i = 0; i < pointNumber; i++) {
+    pcl::PointXYZRGB pt;
```
Comment (kunaltyagi) [inline test/visualization/test_visualization.cpp:67]: Use C++11 random (or PCL wrappers) instead. Eg: There's a cloud generator that might help in this situation

Comment (larshg) [reply test/visualization/test_visualization.cpp:67]: ohh... the work... :D I'll look into it tomorrow maybe 👍

**Comment by kunaltyagi** [comment]
Just a note: There was no test failure. Did the test not run?

**Comment by larshg** [comment]
I updated to use the generator. 

According to logs, the test is build(Ubuntu, OSx), but it was not run.

I'll have a look after this update and if it is still no run, we'll have to investigate...

Windows builds doesn't have VTK so they won't run it.

**Comment by larshg** [comment]
MacOS doesn't run the test_visualizer... So the others probably won't do it either... wierd.

**Comment by taketwo** [comment]
I guess CI does not have `DISPLAY` environment variable defined, so visualization tests are not built: https://github.com/PointCloudLibrary/pcl/blob/917c22c0459df9c4e13e051c17076fb76d0cc461/test/visualization/CMakeLists.txt#L15

**Comment by kunaltyagi** [comment]
Do all visualization tests require a display? If so, we need to use `xvfb` (as a hack if this is true only for some tests)

**Comment by taketwo** [comment]
Well, **all** sounds too serious. There is a single file with two test cases.

**Comment by kunaltyagi** [comment]
I don't know how pcl_vis works 😅 Just adding a virtual buffer for all seemed the fastest way to get the tests running beyond dividing the vis tests in 2 parts.

**Comment by taketwo** [comment]
> Just adding a virtual buffer for all seemed the fastest way to get the tests running

Somebody may give this a try, but I think it's certainly outside of the scope of this PR.

**Comment by kunaltyagi** [comment]
If anyone can confirm the failure induced due to the PR, we can merge #4017

**Comment by taketwo** [comment]
I've tried this PR, but the test passes (Ubuntu 18.04 + VTK 6.3).

**Comment by larshg** [comment]
Maybe its a VTK 8.2 issue only - most users reporting this issue has either been using 8.2 directly or updated to 8.2.

**Comment by larshg** [comment]
The test is set as being build in the [cmake configuration](https://dev.azure.com/PointCloudLibrary/pcl/_build/results?buildId=16109&view=logs&j=ee32b0f4-420e-5974-abf0-bb76a4ab271b&t=6a06a7d0-962a-5421-0000-9b8b322365a4&l=155).

but its not build and run in the [Run Unit Test](https://dev.azure.com/PointCloudLibrary/pcl/_build/results?buildId=16109&view=logs&jobId=d0d530cd-2c5d-5eb2-7bb0-02dfe64fb679&j=ee32b0f4-420e-5974-abf0-bb76a4ab271b&t=11aea1df-f992-554b-c9de-a6b47ae4e2b9).

**Comment by taketwo** [comment]
> The test is set as being build in the cmake configuration.

It's not the test itself, it's the `tests_visualization` module as a whole. (Which contains a single target `test_visualization`).

**Comment by larshg** [comment]
> > The test is set as being build in the cmake configuration.
> 
> It's not the test itself, it's the `tests_visualization` module as a whole. (Which contains a single target `test_visualization`).

Ahh, yes ofc.

**Pull request closed without merging**